### ISSUE TITLE
'/seed/user' not working.

### ISSUE THREAD

**Issue by stevej2608** [ahmerb/typescript-sequelize-example#1]
Running the app creates all the tables successfully.

If I post to `localhost:3000/seed/user` I can see in pgAdmin that the user John Doe has been created.

`1	"John Doe"	"2018-10-08 12:31:25.562+01"	"2018-10-08 12:31:25.562+01"`

From the code I would expect to see the following posts in the db `Posts` table:
```
    posts: [
      {
        name: 'post1',
        title: 'Croissants are tasty',
        text: 'I recently ate a croissant from France. It was nice.',
        category: 'croissants'
      },
      {
        name: 'post2',
        title: 'my fav techno music',
        text: 'I love the song TECHNO by TechnoGang.',
        category: 'techno'
      }
    ]

```
If I inspect the posts table in pgAdmin it's empty. No errors are reported.

Any ideas?

**Comment by stevej2608** [comment]
The following works but is obviously not as concise:
```
app.post('/seed/user', async (req: Request, res: Response) => {

  const user = await db.User.create({name: 'John Doe'});

  const posts = [];

  posts[0] = await db.Post.create({
      name: 'post1',
      title: 'Croissants are tasty',
      text: 'I recently ate a croissant from France. It was nice.',
      category: 'croissants'
    });

  posts[1] = await db.Post.create({
      name: 'post2',
      title: 'my fav techno music',
      text: 'I love the song TECHNO by TechnoGang.',
      category: 'techno'
    })
    
  user.addPosts(posts);
  res.status(201).json({ user });
});
```

**Comment by ahmerb** [comment]
Hey Steve, thanks for noticing this. 😃 

I've been able to reproduce the bug. I've fixed it in efcbb66b102511f2d1b5c5539cc2aaa7dc247f4e. Two things needed to be done to fix this.

First, I couldn't get this to work until I manually added an alias to *every* model association. That is, to any `HasOne`, `BelongsTo`, `BelongsToMany` or `HasMany` calls, it appears you *must* specify an alias in the options object.

```typescript
User.hasMany(models.Comment, { foreignKey: 'AuthorId', as: 'comments' });
```

Or,

```typescript
Comment.belongsTo(models.Post, { as: 'posts' });
```

The `as` option specifies an alias to reference the association by. It also defines what the association methods will end up being called. For example, if I do `Comment.belongsTo(models.User, { as: 'authors' })`, then the methods `comment.getAuthors`, `comment.hasAuthors`, etc will be added by Sequelize.

Secondly, you need to specifiy an `include` option in the `create` call. So,

```typescript
  db.User.create({
    name: 'John Doe',
    posts: [
      {
        name: 'post1',
        title: 'Croissants are tasty',
        text: 'I recently ate a croissant from France. It was nice.',
        category: 'croissants'
      },
      {
        name: 'post2',
        title: 'my fav techno music',
        text: 'I love the song TECHNO by TechnoGang.',
        category: 'techno'
      }
    ]
  },
  // new stuff
  { 
    include: [{ model: db.Post, as: 'posts' }]
  })
```

The `as` option in the `include` option must match the alias you set in the model definition.

Note that, you will probably need to drop your schema in postgres, recreate it and then restart the server. I did this in postgres with

```postgres
drop schema public cascade;
create schema public;
GRANT ALL ON SCHEMA public TO postgres;
```
